boards/arm/nrf52840_mdk/nrf52840_mdk.dts

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (c) 2017 Linaro Limited
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/dts-v1/;
+#include <nordic/nrf52840_qiaa.dtsi>
+
+/ {
+	model = "nRF52840-MDK Dev Kit";
+	compatible = "makerdiary,nrf52840-mdk", "nordic,nrf52840-qiaa",
+		     "nordic,nrf52840";
+
+	chosen {
+		zephyr,console = &uart0;
+#ifdef CONFIG_MCUMGR_SMP_UART
+		zephyr,uart-mcumgr = &uart0;
+#endif
+		zephyr,sram = &sram0;
+		zephyr,flash = &flash0;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		led0_green: led_0 {
+			gpios = <&gpio0 22 GPIO_INT_ACTIVE_LOW>;
+			label = "Green LED 0";
+		};
+		led1_red: led_1 {
+			gpios = <&gpio0 23 GPIO_INT_ACTIVE_LOW>;
+			label = "Red LED 1";
+		};
+		led2_blue: led_2 {
+			gpios = <&gpio0 24 GPIO_INT_ACTIVE_LOW>;
+			label = "Blue LED 2";
+		};
+	};
+
+	buttons {
+		compatible = "gpio-keys";
+		button0: button_0 {
+			gpios = <&gpio1 0 GPIO_PUD_PULL_UP>;
+			label = "Push button switch 0";
+		};
+	};
+
+	/* These aliases are provided for compatibility with samples */
+	aliases {
+		sw0 = &button0;
+		led0 = &led0_green;
+		led1 = &led1_red;
+		led2 = &led2_blue;
+		led0-green = &led0_green;
+		led1-red   = &led1_red;
+		led1-blue  = &led2_blue;
+	};
+};
+
+&adc {
+	status ="ok";
+};
+
+&gpiote {
+	status ="ok";
+};
+
+&gpio0 {
+	status ="ok";
+};
+
+&gpio1 {
+	status ="ok";
+};
+
+&uart0 {
+	compatible = "nordic,nrf-uart";
+	current-speed = <115200>;
+	status = "ok";
+	tx-pin = <20>;
+	rx-pin = <19>;
+	rts-pin = <5>;
+	cts-pin = <7>;
+};
+
+#ifdef CONFIG_UART_1_NRF_UARTE
+&uart1 {
+	current-speed = <115200>;
+	status = "ok";
+	tx-pin = <16>;
+	rx-pin = <15>;
+	rts-pin = <14>;
+	cts-pin = <17>;
+
+#ifdef CONFIG_MODEM_WNCM14A2A
```

Comment:
Do we need this for this board?

Reply:
My guess is that only if you plug a modem into it. It does expose GPIOs, so it belongs here if we think all nRF boards should be compatible with this modem out of the box.